src/shared/seccomp-util.c

```diff
@@ -2079,10 +2079,11 @@ static int seccomp_restrict_sxid(scmp_filter_ctx seccomp, mode_t m) {
         /* The new openat2() system call can't be filtered sensibly, since it moves the flags parameter into
          * an indirect structure. Let's block it entirely for now. That should be a reasonably OK thing to do
          * for now, since openat2() is very new and code generally needs fallback logic anyway to be
-         * compatible with kernels that are not absolutely recent. */
+         * compatible with kernels that are not absolutely recent. We return ENOSYS since this is more likely
+         * to trigger fallback logic than EPERM. */
```

Comment:
comment should say that while usually we return EPERM on policy checks this one isn#t strictly a policy check but a way to force programs into a compat mode so that our policy checks enforced above can take place